--- test/CMakeLists.txt
+++ test/CMakeLists.txt
@@ -15,12 +15,17 @@
 
 # check if we have a Fortran compiler
 include("../cmake/language_support.cmake")
 
 workaround_9220(Fortran EIGEN_Fortran_COMPILER_WORKS)
 
+# workaround gfortran doesn't build on Azure DevOps
+if(WIN32)
+  set(EIGEN_Fortran_COMPILER_WORKS OFF)
+endif()
+
 if(EIGEN_Fortran_COMPILER_WORKS)
   enable_language(Fortran OPTIONAL)
   if(NOT CMAKE_Fortran_COMPILER)
     set(EIGEN_Fortran_COMPILER_WORKS OFF)
   endif()
 endif()
