diff --git a/test/basicstuff.cpp b/test/basicstuff.cpp
index 4ca607c82..46171fd0b 100644
--- a/test/basicstuff.cpp
+++ b/test/basicstuff.cpp
@@ -207,6 +207,10 @@ struct casting_test {
     Matrix<TgtScalar,4,4> n = m.template cast<TgtScalar>();
     for (int i=0; i<m.rows(); ++i) {
       for (int j=0; j<m.cols(); ++j) {
+        std::cerr << "SrcScalar: " << type_name<SrcScalar>() << " TgtScalar: " << type_name<TgtScalar>() << std::endl;
+        std::cerr << "m(i,j)" << m(i, j) << std::endl;
+        std::cerr << "internal::cast<SrcScalar,TgtScalar>(m(i, j)" << internal::cast<SrcScalar,TgtScalar>(m(i, j)) << std::endl;
+        std::cerr << "n(i,j)" << n(i, j) << std::endl;
         VERIFY_IS_APPROX(n(i, j), (internal::cast<SrcScalar,TgtScalar>(m(i, j))));
       }
     }
diff --git a/test/main.h b/test/main.h
index 01667388b..6d2120f1e 100644
--- a/test/main.h
+++ b/test/main.h
@@ -617,6 +617,7 @@ inline bool verifyIsApprox(const Type1& a, const Type2& b)
   bool ret = test_isApprox(a,b);
   if(!ret)
   {
+    std::cerr << "verifyIsApprox: a: " << a << " b: " << b << std::endl;
     std::cerr << "Difference too large wrt tolerance " << get_test_precision(a)  << ", relative error is: " << test_relative_error(a,b) << std::endl;
   }
   return ret;
