--- blas/CMakeLists.txt
+++ blas/CMakeLists.txt
@@ -5,6 +5,8 @@
 
 workaround_9220(Fortran EIGEN_Fortran_COMPILER_WORKS)
 
+set(EIGEN_Fortran_COMPILER_WORKS OFF)
+
 if(EIGEN_Fortran_COMPILER_WORKS)
   enable_language(Fortran OPTIONAL)
   if(NOT CMAKE_Fortran_COMPILER)

--- lapack/CMakeLists.txt
+++ lapack/CMakeLists.txt
@@ -5,6 +5,8 @@
 
 workaround_9220(Fortran EIGEN_Fortran_COMPILER_WORKS)
 
+set(EIGEN_Fortran_COMPILER_WORKS OFF)
+
 if(EIGEN_Fortran_COMPILER_WORKS)
   enable_language(Fortran OPTIONAL)
   if(NOT CMAKE_Fortran_COMPILER)

--- test/CMakeLists.txt
+++ test/CMakeLists.txt
@@ -18,6 +18,8 @@
 
 workaround_9220(Fortran EIGEN_Fortran_COMPILER_WORKS)
 
+set(EIGEN_Fortran_COMPILER_WORKS OFF)
+
 if(EIGEN_Fortran_COMPILER_WORKS)
   enable_language(Fortran OPTIONAL)
   if(NOT CMAKE_Fortran_COMPILER)
